subsystem/src/main/java/org/wildfly/extension/elytron/tls/SSLContextDefinitions.java

```diff
@@ -0,0 +1,2000 @@
+            .setCapabilityReference(PROVIDERS_CAPABILITY, KEY_MANAGER_CAPABILITY)
+            .setAllowExpression(false)
+            .setRestartAllServices()
+            .build();
+
+
+    static final SimpleAttributeDefinition providersTMDefinition = new SimpleAttributeDefinitionBuilder(PROVIDERS)
+            .setCapabilityReference(PROVIDERS_CAPABILITY, TRUST_MANAGER_CAPABILITY)
+            .setAllowExpression(false)
+            .setRestartAllServices()
+            .build();
+
+
+    /** KeyStore definitions **/
+
+    static final SimpleAttributeDefinition TYPE = new SimpleAttributeDefinitionBuilder(Constants.TYPE, ModelType.STRING, true)
+            .setMinSize(1)
+            .setAttributeGroup(Constants.IMPLEMENTATION)
+            .setAllowExpression(false)
+            .setRestartAllServices()
+            .build();
+
+    static final SimpleAttributeDefinition REQUIRED = new SimpleAttributeDefinitionBuilder(Constants.REQUIRED, ModelType.BOOLEAN, true)
+            .setRequires(Constants.PATH)
+            .setAttributeGroup(Constants.FILE)
+            .setDefaultValue(ModelNode.FALSE)
+            .setAllowExpression(true)
+            .setRestartAllServices()
+            .build();
+
+    static final SimpleAttributeDefinition ALIAS_FILTER = new SimpleAttributeDefinitionBuilder(Constants.ALIAS_FILTER, ModelType.STRING, true)
+            .setMinSize(1)
+            .setAllowExpression(true)
+            .setRestartAllServices()
+            .build();
+
+    /* Since XSD doesn't support providing a choice between 2 attributes, validation is done in the definition.
+     * This applies to all types of key stores, key managers, and trust managers
+     * TODO: potentially move validation into XML schema */
+
+    static final SimpleAttributeDefinition KEY_STORE = new SimpleAttributeDefinitionBuilder(Constants.KEY_STORE, ModelType.STRING, true)
+            .setMinSize(1)
+            .setRequired(true)
+            // .setAlternatives(Constants.KEY_STORE_OBJECT)
+            .setFlags(AttributeAccess.Flag.RESTART_RESOURCE_SERVICES)
+            .setAllowExpression(false)
+            .build();
+            
+    static final SimpleAttributeDefinition keystoreKMDefinition = new SimpleAttributeDefinitionBuilder(KEY_STORE)
+            .setCapabilityReference(KEY_STORE_CAPABILITY, KEY_MANAGER_CAPABILITY)
+            .setAllowExpression(false)
+            .setRestartAllServices()
+            .build();
+
+    static final SimpleAttributeDefinition keystoreTMDefinition = new SimpleAttributeDefinitionBuilder(KEY_STORE)
+            .setCapabilityReference(KEY_STORE_CAPABILITY, TRUST_MANAGER_CAPABILITY)
+            .setAllowExpression(false)
+            .setRestartAllServices()
+            .build();
+
+
+    static final ObjectTypeAttributeDefinition KEY_STORE_OBJECT = new ObjectTypeAttributeDefinition.Builder(Constants.KEY_STORE_OBJECT, TYPE,
```

Comment:
@cam-rod  Key-store-object is new resource, right? It will be used for simplification of configuration?

Reply:
yes, just haven't gotten around to implementing them yet